akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowInterleaveSpec.scala

```diff
@@ -44,6 +44,15 @@ class FlowInterleaveSpec extends BaseTwoStreamsSetup {
       probe.expectComplete()
     }
 
+    "complete upstream if any of its upstream completes with eagerClose = true" in assertAllStagesStopped {
+      val probe = TestSubscriber.manualProbe[Int]()
+
+      Source(0 to 2).interleave(Source(3 to 5), 2, eagerClose = true).runWith(Sink.fromSubscriber(probe))
+      probe.expectSubscription().request(10)
+      probe.expectNext(0, 1, 3, 4, 2)
+      probe.expectComplete()
+    }
```

Comment:
> the source that completes does not do it at a segmentSize boundary

What do you mean here? Why shouldn't it complete at a segmentSize boundary? Do you mean, that such code:

`Source(0 to 2).interleave(Source(3 to 5), 3, eagerClose = true)`

shouldn't complete at `Source(0 to 2)` and should return `Source(0 to 5)` instead?